Ets-template-refactor

YanfeiNREL opened the pull request:
#### Because of schema changes, ets-template has to be modified correctly.
#### refactor original ets-templating. 
#### py.test tests/test_translator.py::GeoJSONTranslatorETSTest
Here is the generated ets-template. It is running good in Dymola. 
![001x](https://user-images.githubusercontent.com/43475933/80330003-385a2c80-8801-11ea-811b-bb3b599f6a53.PNG)

nllong commented on tests/model_connectors/test_ets.py:
```diff
@@ -62,11 +59,8 @@ def test_ets_thermal_junction(self):
-
     def test_ets_to_modelica(self):
-        self.assertIsNotNone(self.ets.to_modelica())
+        self.assertIsNotNone(self.ets.ets_to_modelica())
```
this method should just be .to_modelica, not .ets_to_modelica.

nllong commented on tests/model_connectors/test_ets.py:
```diff
@@ -62,11 +59,8 @@ def test_ets_thermal_junction(self):
 
     def test_ets_in_dymola(self):
-        self.assertIsNotNone(self.ets.templated_ets_openloops_dymola())
+        self.assertIsNotNone(self.ets.ets_openloop_to_modelica())
```
If there is a separate method for to_modelica, then perhaps we need to have either a) another ETS class, or b) options on the to_modelica call... I'd prefer the former.

nllong commented on geojson_modelica_translator/geojson/data/schemas/district_system_properties.json:
```diff
@@ -133,6 +133,11 @@
         "Waste Heat Source--Hot Water",
         "Waste Heat Source--Ambient Water"
       ]
+    },
```
If we add content here, then we need to communicate with the URBANopt team as these files are automatically pull from here: https://github.com/urbanopt/urbanopt-geojson-gem/blob/develop/lib/urbanopt/geojson/schema/district_system_properties.json

That is, this file will be overwritten. Probably should go in the system parameter file and schema. To pull in changed run `./setup.py update_schemas`

nllong commented on geojson_modelica_translator/geojson_modelica_translator.py:
```diff
@@ -130,3 +133,22 @@ def to_modelica(self, project_name, save_dir, model_connector_str="TeaserConnect
+        try:
+            ets.check_ets_thermal_junction()
+        except NotImplementedError:
+            print("schema for ets thermal junction has issues!!!")
```
we need to implement some real loggers. No action needed right now, just a comment that this will eventually become _log.info(...)

nllong commented on geojson_modelica_translator/model_connectors/ets_template.py:
```diff
@@ -52,30 +52,17 @@ def __init__(self, thermal_junction_properties_geojson, system_parameters_geojso
-        self.directory_ets_templated = os.path.join(
-            directory_up_two_levels + "/tests/output_ets"
+        self.folder_ets_templated = os.path.join(
+            directory_up_two_levels + "/tests/output/ets"
```
Please update this to not persist anything into the tests directory (unless this is a test).

YanfeiNREL commented:
I updated this branch, and removed scaffold-integration to another branch. This will make  things easier for both of us. . I also revised according to tour comments. Please merge,

nllong commented:
this is not quite ready:
1) update with latest develop
2) do not read in any JSON schemas to obtain values. A schema is the definition of an instance. We need to create an instance of the schema to pass. I'm fixing the schema on this branch here:  https://github.com/urbanopt/geojson-modelica-translator/tree/87-invalid-schema
3) do not write files to the test directory unless you are writing a test.

laurenklun commented:
Cleanup
